### ISSUE TITLE
feature request: lines linking points to underlying raster

### ISSUE THREAD

**Issue by richpauloo** [marcosci/layer#1]
This is a super cool project, and something that I've always wanted to see in rstats. Thank you for thinking of this! 

Take the following with a grain of salt and only if you are inspired by it, or see it as useful. Otherwise, feel free to close the issue.

I'm wondering how difficult it would be to implement a feature to link points or polygon centroids to the underlying raster? Something like the following, but with lines for all points:

![image](https://user-images.githubusercontent.com/10336683/148260653-5dc34c47-0999-4533-a2f5-0ca3ea057ecb.png)

**Comment by marcosci** [comment]
Thanks for your kind words!

Puh. I had the same idea at some point, but my brain was to lazy to come up with a good solution for that.

Since the spatial data itself is shifted, one had to give each feature an ID with respect to its original location. 
After the shifting one could come up with a line feature connecting the same ids and draw them with ggplot2?

Initial thought ... could be possible :D 

Do you want to do this for single features or AOI, say a certain area shown as a rectangle?

**Comment by richpauloo** [comment]
Awesome! 

I'm not sure how to implement the nitty-gritty details, but can imagine a feature like `plot_tiltedmaps(map_list, connect_lines = TRUE)` where by default lines are computed on-the-fly between features and the underlying raster. 

Or to your point of constraining this to a set of features, perhaps `plot_tiltedmaps(map_list, connect_lines = column)` where `column` is a `TRUE/FALSE` column in the points or polygon dataframe that specifies if a line should be drawn for that feature.

**Comment by marcosci** [comment]
Will tackle this as soon as there is some more time :)

**Comment by marcosci** [comment]
I think I have it 👍 Just thinking about the interface now.

![image](https://user-images.githubusercontent.com/10864574/148388150-dc5b8f48-95cf-4e3c-896d-efe7be8cbf12.png)

**Comment by z3tt** [comment]
Yes, here it is cryptic. For me, a good use case would be to indicate a single or only a few points of interest, e.g. mountain peaks. I'd love to see a function and an example, maybe one illustrating it with only a few lines?

**Comment by marcosci** [comment]
@e-kotov maybe you have some input here? The feature was actually quite cool, but we got somehow stuck with a good interface to make it userfriendly.

**Comment by e-kotov** [comment]
@marcosci yes, I had some experiments with this code a few years ago, would be great to implement this. I will try to find the time.

**Comment by e-kotov** [comment]
For anyone interested, some draft implementation of this in https://github.com/marcosci/layer/tree/feat-lines
